### PR TITLE
add ability to override default plural check

### PR DESCRIPTION
**Pull request by jlburkhead**
This is for https://github.com/manyminds/api2go/issues/250

### CODE REVIEW

**Comment by wwwdata** [comment]
lgtm. It will also not crash if no relationship type was specified and default to the `DefaultRelationship`, right? What do you think @sharpner?

**Comment by jlburkhead** [comment]
correct. the zero value of `RelationshipType` is `DefaultRelationship`. I can add a test for that, but I figured it was already implicitly being tested by none of the existing tests failing

**Comment by wwwdata** [comment]
yes, let's wait for our second pr review and wen can merge :) thanks a lot

File: jsonapi/marshal.go [hunk 1]
```diff
@@ -213,7 +234,7 @@ func getStructRelationships(relationer MarshalLinkedRelations, information Serve
 		relationships[name] = Relationship{}
 		// if referenceType is plural, we need to use an array for data, otherwise it's just an object
 		container := RelationshipDataContainer{}
-		if Pluralize(name) == name {
+		if isToMany(referenceIDs[0].Relationship, referenceIDs[0].Name) {
```
Comment (sharpner) [inline jsonapi/marshal.go]: we should check the length of referencedIDs before accessing `0`, could possibly be out of range.

**Comment by sharpner** [comment]
awesome :-) thanks for the PR! this looks really good.

File: jsonapi/marshal.go (hunk 1 above)
Comment (jlburkhead) [inline jsonapi/marshal.go]: `referencedIDs` has to have at least 1 element since `sortedResults` is populated [here](https://github.com/jlburkhead/api2go/blob/75f0e601dcdbfd113d0c6a8c29b6cba3f0b7f480/jsonapi/marshal.go#L222). `referenceIDs[0]` is also accessed [here](https://github.com/jlburkhead/api2go/blob/75f0e601dcdbfd113d0c6a8c29b6cba3f0b7f480/jsonapi/marshal.go#L248) without checking length
